drivers/clock_control/clock_control_mcux_sim.c

```diff
@@ -15,11 +15,19 @@ LOG_MODULE_REGISTER(clock_control);
 
 static int mcux_sim_on(struct device *dev, clock_control_subsys_t sub_system)
 {
+	clock_ip_name_t clock_ip_name = (clock_name_t) sub_system;
+
```

Comment:
The cast of sub_system is incorrect. The sub_system value is actually a clock_ip_name_t which is 32bits in size. The clock_name_t is only 16 bits in size so the cast causes the value to be zero.